### PR TITLE
History pagination and layout fixes

### PR DESCRIPTION
**Pull request by cmgustavo**
* Fix PIN page
* New flow for preferences and `add wallet`
* Manage views from route.js with ui-routes module
* Fixes modals
* History pagination
* Fix generator 's script for devices

### CODE REVIEW

File: src/js/controllers/index.js [hunk 1]
```diff
@@ -42,16 +41,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     go.walletHome();
   });
 
+  $rootScope.$on('newAddress', function(event, addr) {
+    self.addr = addr;
+  });
+
+  $rootScope.$on('updateTransactionHistory', function(event, h) {
+    self.blockchain_txs = h;
```
Comment (matiu) [inline src/js/controllers/index.js]: please use camelCase.

File: src/js/controllers/index.js [hunk 2]
```diff
@@ -42,16 +41,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     go.walletHome();
   });
 
+  $rootScope.$on('newAddress', function(event, addr) {
+    self.addr = addr;
+  });
+
+  $rootScope.$on('updateTransactionHistory', function(event, h) {
```
Comment (matiu) [inline src/js/controllers/index.js]: the other 'update*' event handlers call `fc.get*` functions.

This one only assigns data. It should be called differently.

File: src/js/controllers/receive.js [hunk 3]
```diff
@@ -4,14 +4,22 @@ angular.module('copayApp.controllers').controller('receiveController',
   function($rootScope, $scope, $timeout, $modal, isCordova, isMobile, profileService) {
     var fc = profileService.focusedClient;
 
-    this.showAll = false;
+    this.showAll = false; 
 
-    this.newAddr = function() {
+    this.newAddr = function(addr) {
```
Comment (matiu) [inline src/js/controllers/receive.js]: newAddr
should not receive an addr is confusing.

File: public/views/receive.html [hunk 4]
```diff
@@ -1,19 +1,25 @@
-<div class="addresses" ng-controller="receiveController as receive" ng-init="receive.newAddr()" ng-show="receive.addr">
-
-  <div class="row" ng-if="receive.addr">
+<div class="addresses" ng-controller="receiveController as receive" ng-init="receive.newAddr(index.addr)">
```
Comment (matiu) [inline public/views/receive.html]: see comment below, newAddr should not receive and addr.

**Comment by cmgustavo** [comment]
Thanks a lot for your comments! Updated.

**Pull request merged**